bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/rest/Scenes.java

```diff
@@ -0,0 +1,422 @@
+import org.openhab.io.hueemulation.internal.dto.changerequest.HueChangeSceneEntry;
+import org.openhab.io.hueemulation.internal.dto.changerequest.HueStateChange;
+import org.openhab.io.hueemulation.internal.dto.response.HueResponse;
+import org.openhab.io.hueemulation.internal.dto.response.HueSuccessGeneric;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * Handles Hue scenes via the automation subsystem and the corresponding REST interface
+ *
+ * @author David Graeff - Initial contribution
+ */
+@Component(service = { Scenes.class }, property = "com.eclipsesource.jaxrs.publish=false")
+@NonNullByDefault
+@Path("")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class Scenes implements RegistryChangeListener<Rule> {
+    private final Logger logger = LoggerFactory.getLogger(Scenes.class);
+
+    @Reference
+    protected @NonNullByDefault({}) ConfigStore cs;
+    @Reference
+    protected @NonNullByDefault({}) UserManagement userManagement;
+    @Reference
+    protected @NonNullByDefault({}) RuleRegistry ruleRegistry;
+    @Reference
+    protected @NonNullByDefault({}) ItemRegistry itemRegistry;
+
+    /**
+     * Registers to the {@link RuleRegistry} and enumerates currently existing rules.
+     */
+    @Activate
+    public void activate() {
+        ruleRegistry.removeRegistryChangeListener(this);
+        ruleRegistry.addRegistryChangeListener(this);
+
+        for (Rule item : ruleRegistry.getAll()) {
+            added(item);
+        }
+    }
+
+    @Deactivate
+    public void deactivate() {
+        ruleRegistry.removeRegistryChangeListener(this);
+    }
+
+    @Override
+    public void added(Rule scene) {
+        if (!scene.getTags().contains("scene")) {
+            return;
+        }
+        HueSceneEntry entry = new HueSceneEntry(scene.getName());
+        String desc = scene.getDescription();
+        if (desc != null) {
+            entry.description = desc;
+        }
+
+        List<String> items = new ArrayList<>();
+
+        for (Action a : scene.getActions()) {
+            if (!a.getTypeUID().equals("core.ItemCommandAction")) {
+                continue;
+            }
+            ItemCommandActionConfig config = a.getConfiguration().as(ItemCommandActionConfig.class);
+            Item item;
+            try {
+                item = itemRegistry.getItem(config.itemName);
+            } catch (ItemNotFoundException e) {
+                logger.warn("Rule {} is referring to a non existing item {}", scene.getName(), config.itemName);
```

Comment:
Does the user get any feedback of it this way?

Reply:
Unfortunately not, except from the log statement.
This can happen if a user renames an item, that is referred to in a hue scene.
There is no good way to tell the user about such a problem as long as we do not have a notification infrastructure.